project-folder/model.py

```diff
@@ -1,53 +1,86 @@
+                        zoom_range=0.1, width_shift_range=0.1,
+                        height_shift_range=0.1, horizontal_flip=False,
+                        vertical_flip=False)
 
         # Data
         self.X: np.array
         self.y: np.array
 
         self.model: Sequential
 
+    def split(self):
+        self.X_train, self.X_test, self.y_train, self.y_test =
+        train_test_split(self.X, self.y, test_size=0.20, random_state=42)
+        self.X_train, self.X_validate, self.y_train, self.y_test
+        = train_test_split(self.X_train, self.y_train, test_size=0.20,
+                           random_state=42)
+
     def define_model(self):
         model = Sequential()
-        model.add(Conv2D(32, kernel_size=(3, 3), activation='relu', input_shape=(self.num_rows, self.num_cols, 3)))  # noqa
-        model.add(MaxPooling2D((2, 2)))
-        model.add(Dropout(0.1))
-        model.add(Conv2D(64, (3, 3), activation='relu'))
-        model.add(Dropout(0.3))
+        model.add(Conv2D(32, 3, padding='same', activation='relu',
+                  input_shape=(self.num_rows, self.num_cols, 3)))
+        model.add(Conv2D(32, 3, padding='same', activation='relu'))
+        model.add(MaxPooling2D())
+        model.add(Dropout(0.25))
+
+        model.add(Conv2D(64, 3, padding='same', activation='relu',
+                  input_shape=(self.num_rows, self.num_cols, 3)))
+        model.add(Conv2D(64, 3, padding='same', activation='relu'))
+        model.add(MaxPooling2D())
+        model.add(Dropout(0.4))
+
+        model.add(Conv2D(128, 3, padding='same', activation='relu'))
+        model.add(MaxPooling2D())
+        model.add(Dropout(0.5))
         model.add(Flatten())
-        model.add(Dense(128, activation='relu'))
-        model.add(Dropout(0.3))
-        model.add(Dense(self.num_classes, activation='softmax'))
+        model.add(Dense(512, activation='relu'))
+        model.add(Dropout(0.55))
+        model.add(Dense(7, activation='softmax'))
 
-        model.compile(loss='categorical_crossentropy',
-                      optimizer='adam',
+        model.compile(optimizer='adam', loss='binary_crossentropy',
                       metrics=['accuracy'])
 
-    def reshape_data(self, data):
-        y = data['label']
-        X = data.drop(columns='label')
-        X = np.array(X)
-        self.X = X.reshape(X.shape[0], self.num_rows, self.num_cols, 3)
-        self.y = np.eye(self.num_classes)[np.array(y.astype(int)).reshape(-1)]
-
     # Entry point to training the model
     # Consumes data from DataPipeline
     def model_runner(self, data):
         self.reshape_data(data)
+        self.datagen.fit(self.X_train)
         self.model = self.define_model()
-        self.model.fit(self.X, self.y,
+        self.model.fit(self.X_train, self.y_train,
                        batch_size=self.batch_size,
-                       epochs=self.epochs)
+                       steps_per_epoch=self.X_train.shape[0] //
```

Comment:
you don't need to specify this. The default is number of training data / batch size